### ISSUE TITLE
sometimes search result will be empty in azure app service

### ISSUE THREAD

**Issue by zyhfish** [dnnsoftware/Dnn.Platform#2183]
sometimes search result will be empty in azure app service for some instances, such as if the app have 3 instances, one will not able to search but the other 2 instances works fine. there have error logged:

```
[ERROR] DotNetNuke.Services.Exceptions.Exceptions - System.IO.IOException: An unexpected network error occurred.
 
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.ReadCore(Byte[] buffer, Int32 offset, Int32 count)
   at System.IO.FileStream.Read(Byte[] array, Int32 offset, Int32 count)
   at Lucene.Net.Store.SimpleFSDirectory.SimpleFSIndexInput.ReadInternal(Byte[] b, Int32 offset, Int32 len)
   at Lucene.Net.Store.BufferedIndexInput.Refill()
   at Lucene.Net.Store.BufferedIndexInput.ReadByte()
   at Lucene.Net.Store.IndexInput.ReadVInt()
   at Lucene.Net.Index.TermBuffer.Read(IndexInput input, FieldInfos fieldInfos)
   at Lucene.Net.Index.SegmentTermEnum.Next()
   at Lucene.Net.Index.SegmentTermEnum.ScanTo(Term term)
   at Lucene.Net.Index.TermInfosReader.Get(Term term, Boolean useCache)
   at Lucene.Net.Index.TermInfosReader.Terms(Term term)
   at Lucene.Net.Index.DirectoryReader.MultiTermEnum..ctor(IndexReader topReader, IndexReader[] readers, Int32[] starts, Term t)
   at Lucene.Net.Index.DirectoryReader.Terms(Term term)
   at Lucene.Net.Search.PrefixTermEnum..ctor(IndexReader reader, Term prefix)
   at Lucene.Net.Search.PrefixQuery.GetEnum(IndexReader reader)
   at Lucene.Net.Search.MultiTermQuery.ConstantScoreAutoRewrite.Rewrite(IndexReader reader, MultiTermQuery query)
   at Lucene.Net.Search.BooleanQuery.Rewrite(IndexReader reader)
   at Lucene.Net.Search.BooleanQuery.Rewrite(IndexReader reader)
   at Lucene.Net.Search.BooleanQuery.Rewrite(IndexReader reader)
   at Lucene.Net.Search.IndexSearcher.Rewrite(Query original)
   at Lucene.Net.Search.Query.Weight(Searcher searcher)
   at Lucene.Net.Search.Searcher.Search(Query query, Filter filter, Collector results)
   at DotNetNuke.Services.Search.Internals.LuceneControllerImpl.Search(LuceneSearchContext searchContext)
   at DotNetNuke.Services.Search.Controllers.SearchControllerImpl.GetSecurityTrimmedResults(SearchQuery searchQuery, LuceneQuery luceneQuery)
   at DotNetNuke.Services.Search.Controllers.SearchControllerImpl.GetResults(SearchQuery searchQuery)
   at DotNetNuke.Services.Search.Controllers.SearchControllerImpl.SiteSearch(SearchQuery searchQuery)
   at DotNetNuke.Web.InternalServices.SearchServiceController.GetBasicViews(SearchQuery searchQuery, Int32& totalHits)
   at DotNetNuke.Web.InternalServices.SearchServiceController.GetGroupedBasicViews(SearchQuery query, SearchContentSource userSearchSource, Int32 portalId)
   at DotNetNuke.Web.InternalServices.SearchServiceController.Preview(String keywords, String culture, Int32 forceWild, Int32 portal)
```

the only workaround is restart that instance and search will back to work.

**Comment by tpluscode** [comment]
I'm reopening this because we missed that tests started failing after merging.

**Comment by ohine** [comment]
Closing as this was fixed with #2248
